packages/payment/src/paymentGateway.ts

```diff
@@ -16,4 +17,10 @@ export interface TransferData {
 export interface PaymentGateway {
     prepare(payment: Payment, returnUrl: string): TransferData | PromiseLike<TransferData>;
     postPaymentResult(payment: Payment, resultData: ParsedQueryString): Payment | null | PromiseLike<Payment | null>;
+    receivePaymentNotification(
```

Comment:
Adding this would technically be a breaking change and would require a major version bump since existing implementations don't have this.  Alternatively we can make this optional.  Evoke itself will need to treat it as optional anyway since there may be implementations without this.